src/Passes/Source/ValidationPass/ValidationPass.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Logging/ILogger.hpp"
+#include "Profile/Profile.hpp"
+#include "Types/Types.hpp"
+#include "ValidationPass/ValidationConfiguration.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <functional>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class ValidationPass : public llvm::PassInfoMixin<ValidationPass>
+    {
+      public:
+        using Instruction       = llvm::Instruction;
+        using Value             = llvm::Value;
+        using ConstantArguments = std::unordered_map<std::string, llvm::ConstantInt*>;
+        using ILoggerPtr        = std::shared_ptr<ILogger>;
+
+        // Construction and destruction configuration.
+        //
+
+        explicit ValidationPass(ValidationPassConfiguration const& cfg)
+          : config_{cfg}
+        {
+        }
+
+        /// Copy construction is banned.
+        ValidationPass(ValidationPass const&) = delete;
+
+        /// We allow move semantics.
+        ValidationPass(ValidationPass&&) = default;
+
+        /// Default destruction.
+        ~ValidationPass() = default;
+
+        llvm::PreservedAnalyses run(llvm::Module& module, llvm::ModuleAnalysisManager& mam);
+        /// Whether or not this pass is required to run.
+        static bool isRequired();
```

Comment:
Is this function a requirement of the LLVM mechanisms?

Reply:
Yes, this required by LLVM.

Reply:
(Up to you) Consider adding the explanation to the code comments.